onnxruntime/test/platform/threadpool_test.cc

```diff
@@ -104,6 +104,39 @@ void TestMultipleParallelFor(const std::string& name, int num_threads, int num_c
   }
 }
 
+void TestBurstScheduling(const std::string& name, int num_tasks) {
+  // Test submitting a burst of functions for executing.  The aim is to provoke cases such
+  // as the thread pool's work queues being full.
+  for (int rep = 0; rep < 5; rep++) {
+    std::atomic<int> ctr{0};
+    // Schedule a burst of num_tasks back-to-back, and then cleanly shut down the thread
+    // pool.  The synchronization barrier during shut down should ensure that all of the
+    // tasks are complete.  Note that if the thread pool's work queues are full, then a
+    // call to tp->Schedule() may run its argument synchronously.  In any case, we expect
+    // ctr==num_tasks.
+    CreateThreadPoolAndTest(name, 2, [&](ThreadPool* tp) {
+      // First variant : schedule from outside the pool
+      for (int tasks = 0; tasks < num_tasks; tasks++) {
+        tp->Schedule([&]() {
+          ctr++;
+        });
+      }
+    });
+    ASSERT_TRUE(ctr == num_tasks);
```

Comment:
consider ASSERT_EQ()

Reply:
Thanks -- I see we've got a lot of ASSERT_TRUEs across the test cases, so I'll look at updating this example (and the others) as a separate clean-up PR without substantive changes.